File: android/src/main/java/com/azifacemobile/theme/InitialLoadingAnimation.java
package com.azifacemobile.theme;

import android.graphics.Typeface;

import com.azifacemobile.utils.Theme;
import com.facebook.react.bridge.ReactApplicationContext;

import org.json.JSONObject;

public class InitialLoadingAnimation {
  private final JSONObject target;
  private final Color color;
  private final Font font;

  public InitialLoadingAnimation(ReactApplicationContext context) {
    this.target = new Theme().getTarget("initialLoadingAnimation");
    this.color = new Color();
    this.font = new Font(context);
  }

  public int getTrackColor() {
    return this.color.getColor(this.target, "trackColor", "#b3d4fc");
  }

  public int getFillColor() {
    return this.color.getColor(this.target, "fillColor", "#026ff4");
  }

  public Typeface getFont() {
    return this.font.getTypography(this.target, "font");
  }
}

File: ios/Theme/InitialLoadingAnimation.swift
public class InitialLoadingAnimation {
  private let target: NSDictionary?
  private let color: Color
  private let font: Font
  
  init() {
    self.target = Style().getTarget("initialLoadingAnimation")
    self.color = Color()
    self.font = Font()
  }
  
  func getTrackColor() -> UIColor {
    return self.color.getColor(self.target, key: "trackColor", defaultColor: "b3d4fc")
  }
  
  func getFillColor() -> UIColor {
    return self.color.getColor(self.target, key: "fillColor", defaultColor: "#026ff4")
  }
  
  public func getFont() -> UIFont {
    return self.font.getTypography(theme: self.target, key: "font")
  }
}
